tests/s25Main/UI/testMapSelection.cpp

```diff
@@ -0,0 +1,259 @@
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + freeQuadrantMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == -1);
+    // Click on mission area of first mission => selection will change because mission is playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + firstMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+    // Click on mission area of second mission => selection will not change because mission is not playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + secondMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+    // Click on mission area of third mission => selection will not change because mission is not playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + thirdMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+
+    // Activate first and second mission
+    ms.setMissionsStatus({{true, false}, {true, false}, {false, false}});
+    // Selection stays the same as before
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+    // Click on mission area of second mission => selection will change because mission is playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + secondMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 1);
+    // Click on mission area of third mission => selection will not change because mission is not playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + thirdMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 1);
+    // Click on non mission area => selection will not change
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + freeQuadrantMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 1);
+    // Click on mission area of first mission => selection will change because mission is playable
+    ms.Msg_LeftUp(MouseCoords(ms.GetPos() + offsetOfBackground + firstMissionMidPoint));
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+
+    // Test selection
+    // Activate first and second mission
+    ms.setMissionsStatus({{true, false}, {true, false}, {false, false}});
+    // Setting selection to third mission not possible because mission is not playable => stay at previous value
+    ms.setSelection(2);
+    BOOST_TEST(ms.getCurrentSelection() == 0);
+    // Setting selection to second mission is possible because mission is playable
+    ms.setSelection(1);
+    BOOST_TEST(ms.getCurrentSelection() == 1);
+    // Setting selection to invalid mission resets selection to not set
+    ms.setSelection(3);
+    BOOST_TEST(ms.getCurrentSelection() == -1);
+    // Setting selection to third mission not possible because mission is not playable => stay at previous value
+    ms.setSelection(2);
+    BOOST_TEST(ms.getCurrentSelection() == -1);
+    // Setting selection to second mission is possible because mission is playable
+    ms.setSelection(1);
+    BOOST_TEST(ms.getCurrentSelection() == 1);
```

Comment:
Is it on purpose that you check the 2nd mission twice and the first not at all? Did you intend:
```suggestion
    // Setting selection to first mission is possible because mission is playable
    ms.setSelection(0);
    BOOST_TEST(ms.getCurrentSelection() == 0);
```